enwaytech/livox_ros_driver2
# IMU driver: follow ROS msg specs, set covariances

## Additions & Changes
Several modifications to the Livox driver related to the internal IMU driver and data publishing.
These changes are required for the robot_localization's EKF to use the IMU data.

Note: The gyroscope bias compensation and the external frame transformation were moved to the main repo. See PR https://github.com/enwaytech/cleansquare_ros/pull/4267.

- set orientation covariance following the ROS msg's specs to show no orientation data is available (1st element of covariance vector should be -1)
- Convert the linear acceleration units from g's to m/s2, following the ROS msg specs
- add covariances from config for linear acceleration and angular velocity
- Change the IMU frame to take into account the distance between the LIDAR center and the IMU center (translation taken from Livox user manual)
- make sure driver fails if the user_config parsing failed

Reverted:
- ~~Add an option to transform the IMU data to an external frame before publishing. This is needed as the EKF expects the IMU data to be in a frame with the same orientation as base_link.~~
- ~~Add an option to compensate gyroscope bias using static bias values. This is required since the Livox IMUs show high values of bias, especially for the angular velocity around x (in the order of 2-3 degrees/sec)~~


## How to test & expected outcomes
> **Note to the reviewer:** The instructions below might not be complete. If you think more tests are necessary, please do not hesitate to perform them and to record the result in a comment in this PR.

Test together with the main repo PR https://github.com/enwaytech/cleansquare_ros/pull/4267

Run on the robot and check that all the modifications took place (covariances, acceleration units, ~~frame transform, gyro bias,~~ ...)
You can plot the angular velocities in rqt_plot and make sure they are close to 0 over time (ideally, they should be in the order of 0.00X)

## Recommended reviewer
Georg?

### Checklist (tick off when done or remove when not relevant):
- [x] Set labels (do not forget a review time estimation label)
- [x] Add package dependencies
- [x] Check the linter warnings in the PR checks or [locally](https://enwaytech.atlassian.net/wiki/spaces/ENWAY/pages/5308462/C+Coding+Guideline#Code-Checking:-Local-Lint-Reports)


[sc-14152]
[sc-14159]
[sc-14176]
